src/global_timestamp_reader.cpp

```diff
@@ -161,23 +193,21 @@ namespace librealsense
         {
             if (!_users_count)
                 throw wrong_api_call_sequence_exception("time_diff_keeper::update_diff_time called before object started.");
-            std::lock_guard<std::recursive_mutex> lock(_mtx);
             double system_time_start = duration<double, std::milli>(system_clock::now().time_since_epoch()).count();
 
             double sample_hw_time = _device->get_device_time_ms();
             double system_time_finish = duration<double, std::milli>(system_clock::now().time_since_epoch()).count();
             if (system_time_finish - system_time_start > 2.0)
                 throw io_exception("get_device_time_ms() took too long (more then 2 mSecs)");
```

Comment:
Can be added to log as error/warning. No need for exception here.

Reply:
I seems cleaner to use exception then LOG +"return false" since already inside a try-catch section.

Reply:
This is a recoverable case which does not necessary requires user interaction, as the system can recover from it automatically. 
Adding it to log is needed in any case.
Also - I noticed in the latest v2.34.0 under linux the message appears much often than 2 sec in Viewer.